cmd/analyze.go

```diff
@@ -39,22 +39,20 @@ var analyzeCmd = &cobra.Command{
 	Long: `
 `,
 	Example: "  harpoon analyze --exclude vendor/ /path/to/repo/",
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
 		if exclude != "" {
 			excludedPaths = strings.Split(exclude, ",")
 		}
 
 		file, err := os.Open("go.mod")
 		if err != nil {
-			fmt.Printf("failed to open %s: %v\n", "go.mod", err)
-			return
+			return fmt.Errorf("failed to open %s: %w", "go.mod", err)
 		}
 		defer file.Close()
 
 		moduleName, err := analyzer.GetModuleName(file)
 		if err != nil {
-			fmt.Println("module name not found in go.mod")
-			return
+			return fmt.Errorf("module name not found in go.mod")
```

Comment:
```suggestion
			return errors.New("module name not found in go.mod")
```

No sprintf formatting here